modules/tests/test_opendaq_device_modules/test_native_streaming_modules.cpp

```diff
@@ -607,3 +628,119 @@ TEST_F(NativeStreamingModulesTest, GetConfigurationConnectionInfo)
+
+    auto reader = daq::StreamReader<double, uint64_t>(signal, ReadTimeoutType::Any);
+    ASSERT_FALSE(test_helpers::waitForAcknowledgement(signalSubscribeFuture, std::chrono::seconds(1)));
+}
+
+
+TEST_F(NativeStreamingModulesTest, ProtectedSignalSubscribeAllowed)
+{
+    auto users = List<IUser>();
+    users.pushBack(User("admin", "admin", {"admin"}));
+    auto authenticationProvider = StaticAuthenticationProvider(false, users);
+
+    auto server = CreateServerInstance(authenticationProvider);
+    auto client = CreateClientInstance("admin", "admin");
+
+    auto signal = client.getSignalsRecursive()[0].template asPtr<IMirroredSignalConfig>();
+
+    // allow admin on signal
+    auto permissionsAllowAdmin =
+        PermissionsBuilder().inherit(false).assign("admin", PermissionMaskBuilder().read().write().execute()).build();
+    auto serverSignal = server.getSignalsRecursive()[0];
+    serverSignal.getPermissionManager().setPermissions(permissionsAllowAdmin);
+
+    // subscribe
+    std::promise<StringPtr> signalSubscribePromise;
+    std::future<StringPtr> signalSubscribeFuture;
+    test_helpers::setupSubscribeAckHandler(signalSubscribePromise, signalSubscribeFuture, signal);
+
+    auto reader = daq::StreamReader<double, uint64_t>(signal, ReadTimeoutType::Any);
+    ASSERT_TRUE(test_helpers::waitForAcknowledgement(signalSubscribeFuture));
+
+    // unsibscribe
+    std::promise<StringPtr> signalUnsubscribePromise;
+    std::future<StringPtr> signalUnsubscribeFuture;
+    test_helpers::setupUnsubscribeAckHandler(signalUnsubscribePromise, signalUnsubscribeFuture, signal);
+
+    reader.release();
+    ASSERT_TRUE(test_helpers::waitForAcknowledgement(signalUnsubscribeFuture));
+}
+
+TEST_F(NativeStreamingModulesTest, ProtectedSignalUnsubscribeDenied)
+{
+    auto server = CreateServerInstance();
+    auto client = CreateClientInstance();
+
+    auto signal = client.getSignalsRecursive()[0].template asPtr<IMirroredSignalConfig>();
+
+    // subscribe
+    std::promise<StringPtr> signalSubscribePromise;
+    std::future<StringPtr> signalSubscribeFuture;
+    test_helpers::setupSubscribeAckHandler(signalSubscribePromise, signalSubscribeFuture, signal);
+
+    auto reader = daq::StreamReader<double, uint64_t>(signal, ReadTimeoutType::Any);
+    ASSERT_TRUE(test_helpers::waitForAcknowledgement(signalSubscribeFuture));
+
+    // deny everyone on signal
+    auto permissionsDenyAll = PermissionsBuilder().inherit(false).build();
+    auto serverSignal = server.getSignalsRecursive()[0];
+    serverSignal.getPermissionManager().setPermissions(permissionsDenyAll);
+
+    // try to unsibscibe
+    std::promise<StringPtr> signalUnsubscribePromise;
+    std::future<StringPtr> signalUnsubscribeFuture;
+    test_helpers::setupUnsubscribeAckHandler(signalUnsubscribePromise, signalUnsubscribeFuture, signal);
+
+    reader.release();
+    ASSERT_FALSE(test_helpers::waitForAcknowledgement(signalUnsubscribeFuture, std::chrono::seconds(1)));
+}
```

Comment:
I don't think sending a signal available event if user access was revoked after connect is even needed. At least not for now. The only reason, I am revoking read access on the signal after connect is because I want to test if the subscribe method itself is protected. If I were to deny a read access before connect, the client would not even see the signal, so testing the subscribe through standard interface would be impossible.